torchvision/prototype/features/_feature.py

```diff
@@ -28,13 +29,14 @@ def __new__(
         device: Optional[Union[torch.device, str, int]] = None,
         requires_grad: bool = False,
     ) -> F:
-        return cast(
-            F,
-            torch.Tensor._make_subclass(
-                cast(_TensorBase, cls),
-                torch.as_tensor(data, dtype=dtype, device=device),  # type: ignore[arg-type]
-                requires_grad,
-            ),
+        return (
+            torch.as_tensor(  # type: ignore[return-value]
+                data,
+                dtype=dtype,  # type: ignore[arg-type]
+                device=device,  # type: ignore[arg-type]
+            )
```

Comment:
Do I understand correctly that here we use `as_tensor` only to ensure that the requested `dtype` and `device` are respected? Otherwise we could just be doing `data.as_subclass(cls)` because we don't have cases were `data` are numpy array. I wonder if that's faster than `data.to(dtype=dtype, device=device).as_subclass(cls)`?

Reply:
Data is not always a tensor as far as I understand

Reply:
Correct. Basically, the constructor of features currently accepts everything that can be turned into a tensor. This is on par with what `torch.Tensor` does, i.e

```pycon
>>> torch.Tensor([0, 1, 2])
tensor([0., 1., 2.])
```

Reply:
Doing `to()` seems faster.

I use this approach to measure:
```python
import timeit
import torch

data = torch.randint(0, 256, (32, 3, 128, 128))
n = 10000


print(f"Torch version: {torch.__version__}")
for device in [None, "cpu", "cuda"]:
    for dtype in [None, torch.uint8, torch.float16, torch.float32]:
        as_tensor_time = timeit.timeit(lambda: torch.as_tensor(data, device=device, dtype=dtype), number=n)
        to_time = timeit.timeit(lambda: data.to(device=device, dtype=dtype), number=n)
        winner = "as_tensor" if as_tensor_time < to_time else "to"
        print(f"Winner: {winner}(device={device}, dtype={dtype}) - as_tensor: {as_tensor_time}, to: {to_time}")
```

I'm having issues with our cluster and had to run it on an old PyTorch version but here is the output:
```
Torch version: 1.11.0.dev20211102
Winner: to(device=None, dtype=None) - as_tensor: 0.0055221691727638245, to: 0.003016095608472824
Winner: to(device=None, dtype=torch.uint8) - as_tensor: 0.9256818778812885, to: 0.9209173992276192
Winner: to(device=None, dtype=torch.float16) - as_tensor: 3.5527884736657143, to: 3.2720518335700035
Winner: to(device=None, dtype=torch.float32) - as_tensor: 0.6391827426850796, to: 0.6276023834943771
Winner: to(device=cpu, dtype=None) - as_tensor: 0.011921301484107971, to: 0.008555687963962555
Winner: to(device=cpu, dtype=torch.uint8) - as_tensor: 0.6593559384346008, to: 0.6507497057318687
Winner: to(device=cpu, dtype=torch.float16) - as_tensor: 2.140716001391411, to: 2.108065687119961
Winner: to(device=cpu, dtype=torch.float32) - as_tensor: 0.6577045619487762, to: 0.6373300738632679
Winner: to(device=cuda, dtype=None) - as_tensor: 13.870690949261189, to: 11.862402316182852
Winner: to(device=cuda, dtype=torch.uint8) - as_tensor: 4.089442141354084, to: 3.84384398534894
Winner: to(device=cuda, dtype=torch.float16) - as_tensor: 7.160290367901325, to: 7.144624650478363
Winner: to(device=cuda, dtype=torch.float32) - as_tensor: 10.704698234796524, to: 10.372353758662939
```

Reply:
As discussed offline, it is useful to keep the `torch.as_tensor` call to enable things like 

```py
label = features.Label(0)
```